packages/kbn-config-schema/src/types/array_type.ts

```diff
@@ -31,7 +31,7 @@ export class ArrayType<T> extends Type<T[]> {
     let schema = internals
       .array()
       .items(type.getSchema().optional())
-      .sparse();
+      .sparse(false);
```

Comment:
We could add this as an option in  `ArrayOptions`, however 
- I agree that I really don't see the need of sparse arrays
- It would be difficult to adapt the current `ArrayType<T> extends Type<T[]>` to properly reflect this options's value.